src/GrammarDecl.js

```diff
@@ -80,6 +85,11 @@ GrammarDecl.prototype.withDefaultStartRule = function(ruleName) {
   return this;
 };
 
+GrammarDecl.prototype.withSource = function(source) {
+  this.inputStream = InputStream.newFor(source);
```

Comment:
It's strange for `GrammarDecl` to have an `InputStream` when what you really want is an `Interval`. Why not replace this line with:

```this.interval = new Interval(source, 0, source.length);```